From e367aef1a724b2bdcf750fb846af366d7bc49441 Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Wed, 6 Oct 2021 11:44:31 +0200
Subject: [PATCH] lxd/sys/os: reorder kernel features

Features like "CloseRange" are pure kernel features, i.e. they don't
require checking the LXC shared library for support. Others such as
PidFds currently require a supporting LXC library version.
Move them into separate alignment blocks.

Some entries like "CoreScheduling" shouldn't depend on LXC library
version as well as kernel version in order to be marked as available
since they are useful to both container and vms and vms aren't limited
by LXC version obviously.

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 lxd/sys/os.go | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/lxd/sys/os.go b/lxd/sys/os.go
index 99204ec91..2688c2ea5 100644
--- a/lxd/sys/os.go
+++ b/lxd/sys/os.go
@@ -73,18 +73,19 @@ type OS struct {
 
 	// Kernel features
 	CloseRange              bool
-	CoreScheduling          bool
-	NativeTerminals         bool
 	NetnsGetifaddrs         bool
-	PidFds                  bool
 	PidFdSetns              bool
 	SeccompListener         bool
-	SeccompListenerAddfd    bool
 	SeccompListenerContinue bool
 	Shiftfs                 bool
 	UeventInjection         bool
 	VFS3Fscaps              bool
 
+	CoreScheduling       bool
+	NativeTerminals      bool
+	PidFds               bool
+	SeccompListenerAddfd bool
+
 	// LXC features
 	LXCFeatures map[string]bool
 }
-- 
2.33.1

